File: samples/client/wordnik/scala/src/main/scala/io/swagger/client/api/WordApi.scala
package io.swagger.client.api

import scala.collection.mutable.HashMap

class WordApi(val defBasePath: String = "https://api.wordnik.com/v4",
              defApiInvoker: ApiInvoker = ApiInvoker) {
  var basePath = defBasePath
  var apiInvoker = defApiInvoker

  def addHeader(key: String, value: String) = apiInvoker.defaultHeaders += key -> value


  def getWord(word: String, useCanonical: String, includeSuggestions: String): Option[WordObject] = {
    // create path and map variables
    val path = "/word.json/{word}".replaceAll("\\{format\\}", "json").replaceAll("\\{" + "word" + "\\}", apiInvoker.escape(word))




    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(useCanonical) != "null") queryParams += "useCanonical" -> useCanonical.toString
    if (String.valueOf(includeSuggestions) != "null") queryParams += "includeSuggestions" -> includeSuggestions.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[WordObject]).asInstanceOf[WordObject])

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def getAudio(word: String, useCanonical: String, limit: Integer): Option[List[AudioFile]] = {
    // create path and map variables
    val path = "/word.json/{word}/audio".replaceAll("\\{format\\}", "json").replaceAll("\\{" + "word" + "\\}", apiInvoker.escape(word))




    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(useCanonical) != "null") queryParams += "useCanonical" -> useCanonical.toString
    if (String.valueOf(limit) != "null") queryParams += "limit" -> limit.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "array", classOf[AudioFile]).asInstanceOf[List[AudioFile]])

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def getDefinitions(word: String, limit: Integer, partOfSpeech: String, includeRelated: String, sourceDictionaries: List[String], useCanonical: String, includeTags: String): Option[List[Definition]] = {
    // create path and map variables
    val path = "/word.json/{word}/definitions".replaceAll("\\{format\\}", "json").replaceAll("\\{" + "word" + "\\}", apiInvoker.escape(word))




    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(limit) != "null") queryParams += "limit" -> limit.toString
    if (String.valueOf(partOfSpeech) != "null") queryParams += "partOfSpeech" -> partOfSpeech.toString
    if (String.valueOf(includeRelated) != "null") queryParams += "includeRelated" -> includeRelated.toString
    if (String.valueOf(sourceDictionaries) != "null") queryParams += "sourceDictionaries" -> sourceDictionaries.toString
    if (String.valueOf(useCanonical) != "null") queryParams += "useCanonical" -> useCanonical.toString
    if (String.valueOf(includeTags) != "null") queryParams += "includeTags" -> includeTags.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "array", classOf[Definition]).asInstanceOf[List[Definition]])

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def getEtymologies(word: String, useCanonical: String): Option[List[String]] = {
    // create path and map variables
    val path = "/word.json/{word}/etymologies".replaceAll("\\{format\\}", "json").replaceAll("\\{" + "word" + "\\}", apiInvoker.escape(word))




    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(useCanonical) != "null") queryParams += "useCanonical" -> useCanonical.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "array", classOf[String]).asInstanceOf[List[String]])

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def getExamples(word: String, includeDuplicates: String, useCanonical: String, skip: Integer, limit: Integer) = {
    // create path and map variables
    val path = "/word.json/{word}/examples".replaceAll("\\{format\\}", "json").replaceAll("\\{" + "word" + "\\}", apiInvoker.escape(word))




    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(includeDuplicates) != "null") queryParams += "includeDuplicates" -> includeDuplicates.toString
    if (String.valueOf(useCanonical) != "null") queryParams += "useCanonical" -> useCanonical.toString
    if (String.valueOf(skip) != "null") queryParams += "skip" -> skip.toString
    if (String.valueOf(limit) != "null") queryParams += "limit" -> limit.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def getWordFrequency(word: String, useCanonical: String, startYear: Integer, endYear: Integer): Option[FrequencySummary] = {
    // create path and map variables
    val path = "/word.json/{word}/frequency".replaceAll("\\{format\\}", "json").replaceAll("\\{" + "word" + "\\}", apiInvoker.escape(word))




    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(useCanonical) != "null") queryParams += "useCanonical" -> useCanonical.toString
    if (String.valueOf(startYear) != "null") queryParams += "startYear" -> startYear.toString
    if (String.valueOf(endYear) != "null") queryParams += "endYear" -> endYear.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[FrequencySummary]).asInstanceOf[FrequencySummary])

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def getHyphenation(word: String, useCanonical: String, sourceDictionary: String, limit: Integer) = {
    // create path and map variables
    val path = "/word.json/{word}/hyphenation".replaceAll("\\{format\\}", "json").replaceAll("\\{" + "word" + "\\}", apiInvoker.escape(word))




    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(useCanonical) != "null") queryParams += "useCanonical" -> useCanonical.toString
    if (String.valueOf(sourceDictionary) != "null") queryParams += "sourceDictionary" -> sourceDictionary.toString
    if (String.valueOf(limit) != "null") queryParams += "limit" -> limit.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def getPhrases(word: String, limit: Integer, wlmi: Integer, useCanonical: String): Option[List[Bigram]] = {
    // create path and map variables
    val path = "/word.json/{word}/phrases".replaceAll("\\{format\\}", "json").replaceAll("\\{" + "word" + "\\}", apiInvoker.escape(word))




    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(limit) != "null") queryParams += "limit" -> limit.toString
    if (String.valueOf(wlmi) != "null") queryParams += "wlmi" -> wlmi.toString
    if (String.valueOf(useCanonical) != "null") queryParams += "useCanonical" -> useCanonical.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "array", classOf[Bigram]).asInstanceOf[List[Bigram]])

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def getTextPronunciations(word: String, useCanonical: String, sourceDictionary: String, typeFormat: String, limit: Integer) = {
    // create path and map variables
    val path = "/word.json/{word}/pronunciations".replaceAll("\\{format\\}", "json").replaceAll("\\{" + "word" + "\\}", apiInvoker.escape(word))




    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(useCanonical) != "null") queryParams += "useCanonical" -> useCanonical.toString
    if (String.valueOf(sourceDictionary) != "null") queryParams += "sourceDictionary" -> sourceDictionary.toString
    if (String.valueOf(typeFormat) != "null") queryParams += "typeFormat" -> typeFormat.toString
    if (String.valueOf(limit) != "null") queryParams += "limit" -> limit.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def getRelatedWords(word: String, useCanonical: String, relationshipTypes: String, limitPerRelationshipType: Integer) = {
    // create path and map variables
    val path = "/word.json/{word}/relatedWords".replaceAll("\\{format\\}", "json").replaceAll("\\{" + "word" + "\\}", apiInvoker.escape(word))




    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(useCanonical) != "null") queryParams += "useCanonical" -> useCanonical.toString
    if (String.valueOf(relationshipTypes) != "null") queryParams += "relationshipTypes" -> relationshipTypes.toString
    if (String.valueOf(limitPerRelationshipType) != "null") queryParams += "limitPerRelationshipType" -> limitPerRelationshipType.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def getTopExample(word: String, useCanonical: String): Option[Example] = {
    // create path and map variables
    val path = "/word.json/{word}/topExample".replaceAll("\\{format\\}", "json").replaceAll("\\{" + "word" + "\\}", apiInvoker.escape(word))




    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(useCanonical) != "null") queryParams += "useCanonical" -> useCanonical.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[Example]).asInstanceOf[Example])

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

}

File: samples/client/wordnik/java/src/main/java/io/swagger/client/api/WordApi.java
package io.swagger.client.api;

import com.sun.jersey.multipart.FormDataMultiPart;
import io.swagger.client.ApiException;
import io.swagger.client.ApiInvoker;
import io.swagger.client.model.AudioFile;
import io.swagger.client.model.Bigram;
import io.swagger.client.model.Definition;
import io.swagger.client.model.Example;
import io.swagger.client.model.FrequencySummary;
import io.swagger.client.model.WordObject;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class WordApi {
    String basePath = "https://api.wordnik.com/v4";
    ApiInvoker apiInvoker = ApiInvoker.getInstance();

    public ApiInvoker getInvoker() {
        return apiInvoker;
    }

    public String getBasePath() {
        return basePath;
    }

    public void setBasePath(String basePath) {
        this.basePath = basePath;
    }

    public WordObject getWord(String word, String useCanonical, String includeSuggestions) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/word.json/{word}".replaceAll("\\{format\\}", "json")
                .replaceAll("\\{" + "word" + "\\}", apiInvoker.escapeString(word.toString()));

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(useCanonical))) {
            queryParams.put("useCanonical", String.valueOf(useCanonical));
        }
        if (!"null".equals(String.valueOf(includeSuggestions))) {
            queryParams.put("includeSuggestions", String.valueOf(includeSuggestions));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return (WordObject) ApiInvoker.deserialize(response, "", WordObject.class);
            } else {
                return null;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return null;
            } else {
                throw ex;
            }
        }
    }


    public List<AudioFile> getAudio(String word, String useCanonical, Integer limit) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/word.json/{word}/audio".replaceAll("\\{format\\}", "json")
                .replaceAll("\\{" + "word" + "\\}", apiInvoker.escapeString(word.toString()));

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(useCanonical))) {
            queryParams.put("useCanonical", String.valueOf(useCanonical));
        }
        if (!"null".equals(String.valueOf(limit))) {
            queryParams.put("limit", String.valueOf(limit));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return (List<AudioFile>) ApiInvoker.deserialize(response, "array", AudioFile.class);
            } else {
                return null;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return null;
            } else {
                throw ex;
            }
        }
    }


    public List<Definition> getDefinitions(String word, Integer limit, String partOfSpeech, String includeRelated, List<String> sourceDictionaries, String useCanonical, String includeTags) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/word.json/{word}/definitions".replaceAll("\\{format\\}", "json")
                .replaceAll("\\{" + "word" + "\\}", apiInvoker.escapeString(word.toString()));

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(limit))) {
            queryParams.put("limit", String.valueOf(limit));
        }
        if (!"null".equals(String.valueOf(partOfSpeech))) {
            queryParams.put("partOfSpeech", String.valueOf(partOfSpeech));
        }
        if (!"null".equals(String.valueOf(includeRelated))) {
            queryParams.put("includeRelated", String.valueOf(includeRelated));
        }
        if (!"null".equals(String.valueOf(sourceDictionaries))) {
            queryParams.put("sourceDictionaries", String.valueOf(sourceDictionaries));
        }
        if (!"null".equals(String.valueOf(useCanonical))) {
            queryParams.put("useCanonical", String.valueOf(useCanonical));
        }
        if (!"null".equals(String.valueOf(includeTags))) {
            queryParams.put("includeTags", String.valueOf(includeTags));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return (List<Definition>) ApiInvoker.deserialize(response, "array", Definition.class);
            } else {
                return null;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return null;
            } else {
                throw ex;
            }
        }
    }


    public List<String> getEtymologies(String word, String useCanonical) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/word.json/{word}/etymologies".replaceAll("\\{format\\}", "json")
                .replaceAll("\\{" + "word" + "\\}", apiInvoker.escapeString(word.toString()));

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(useCanonical))) {
            queryParams.put("useCanonical", String.valueOf(useCanonical));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return (List<String>) ApiInvoker.deserialize(response, "array", String.class);
            } else {
                return null;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return null;
            } else {
                throw ex;
            }
        }
    }


    public void getExamples(String word, String includeDuplicates, String useCanonical, Integer skip, Integer limit) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/word.json/{word}/examples".replaceAll("\\{format\\}", "json")
                .replaceAll("\\{" + "word" + "\\}", apiInvoker.escapeString(word.toString()));

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(includeDuplicates))) {
            queryParams.put("includeDuplicates", String.valueOf(includeDuplicates));
        }
        if (!"null".equals(String.valueOf(useCanonical))) {
            queryParams.put("useCanonical", String.valueOf(useCanonical));
        }
        if (!"null".equals(String.valueOf(skip))) {
            queryParams.put("skip", String.valueOf(skip));
        }
        if (!"null".equals(String.valueOf(limit))) {
            queryParams.put("limit", String.valueOf(limit));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return;
            } else {
                return;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return;
            } else {
                throw ex;
            }
        }
    }


    public FrequencySummary getWordFrequency(String word, String useCanonical, Integer startYear, Integer endYear) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/word.json/{word}/frequency".replaceAll("\\{format\\}", "json")
                .replaceAll("\\{" + "word" + "\\}", apiInvoker.escapeString(word.toString()));

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(useCanonical))) {
            queryParams.put("useCanonical", String.valueOf(useCanonical));
        }
        if (!"null".equals(String.valueOf(startYear))) {
            queryParams.put("startYear", String.valueOf(startYear));
        }
        if (!"null".equals(String.valueOf(endYear))) {
            queryParams.put("endYear", String.valueOf(endYear));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return (FrequencySummary) ApiInvoker.deserialize(response, "", FrequencySummary.class);
            } else {
                return null;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return null;
            } else {
                throw ex;
            }
        }
    }


    public void getHyphenation(String word, String useCanonical, String sourceDictionary, Integer limit) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/word.json/{word}/hyphenation".replaceAll("\\{format\\}", "json")
                .replaceAll("\\{" + "word" + "\\}", apiInvoker.escapeString(word.toString()));

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(useCanonical))) {
            queryParams.put("useCanonical", String.valueOf(useCanonical));
        }
        if (!"null".equals(String.valueOf(sourceDictionary))) {
            queryParams.put("sourceDictionary", String.valueOf(sourceDictionary));
        }
        if (!"null".equals(String.valueOf(limit))) {
            queryParams.put("limit", String.valueOf(limit));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return;
            } else {
                return;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return;
            } else {
                throw ex;
            }
        }
    }


    public List<Bigram> getPhrases(String word, Integer limit, Integer wlmi, String useCanonical) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/word.json/{word}/phrases".replaceAll("\\{format\\}", "json")
                .replaceAll("\\{" + "word" + "\\}", apiInvoker.escapeString(word.toString()));

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(limit))) {
            queryParams.put("limit", String.valueOf(limit));
        }
        if (!"null".equals(String.valueOf(wlmi))) {
            queryParams.put("wlmi", String.valueOf(wlmi));
        }
        if (!"null".equals(String.valueOf(useCanonical))) {
            queryParams.put("useCanonical", String.valueOf(useCanonical));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return (List<Bigram>) ApiInvoker.deserialize(response, "array", Bigram.class);
            } else {
                return null;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return null;
            } else {
                throw ex;
            }
        }
    }


    public void getTextPronunciations(String word, String useCanonical, String sourceDictionary, String typeFormat, Integer limit) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/word.json/{word}/pronunciations".replaceAll("\\{format\\}", "json")
                .replaceAll("\\{" + "word" + "\\}", apiInvoker.escapeString(word.toString()));

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(useCanonical))) {
            queryParams.put("useCanonical", String.valueOf(useCanonical));
        }
        if (!"null".equals(String.valueOf(sourceDictionary))) {
            queryParams.put("sourceDictionary", String.valueOf(sourceDictionary));
        }
        if (!"null".equals(String.valueOf(typeFormat))) {
            queryParams.put("typeFormat", String.valueOf(typeFormat));
        }
        if (!"null".equals(String.valueOf(limit))) {
            queryParams.put("limit", String.valueOf(limit));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return;
            } else {
                return;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return;
            } else {
                throw ex;
            }
        }
    }


    public void getRelatedWords(String word, String useCanonical, String relationshipTypes, Integer limitPerRelationshipType) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/word.json/{word}/relatedWords".replaceAll("\\{format\\}", "json")
                .replaceAll("\\{" + "word" + "\\}", apiInvoker.escapeString(word.toString()));

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(useCanonical))) {
            queryParams.put("useCanonical", String.valueOf(useCanonical));
        }
        if (!"null".equals(String.valueOf(relationshipTypes))) {
            queryParams.put("relationshipTypes", String.valueOf(relationshipTypes));
        }
        if (!"null".equals(String.valueOf(limitPerRelationshipType))) {
            queryParams.put("limitPerRelationshipType", String.valueOf(limitPerRelationshipType));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return;
            } else {
                return;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return;
            } else {
                throw ex;
            }
        }
    }


    public Example getTopExample(String word, String useCanonical) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/word.json/{word}/topExample".replaceAll("\\{format\\}", "json")
                .replaceAll("\\{" + "word" + "\\}", apiInvoker.escapeString(word.toString()));

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(useCanonical))) {
            queryParams.put("useCanonical", String.valueOf(useCanonical));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return (Example) ApiInvoker.deserialize(response, "", Example.class);
            } else {
                return null;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return null;
            } else {
                throw ex;
            }
        }
    }

}
